tensorboard/webapp/widgets/linked_time_fob/types.ts

```diff
@@ -0,0 +1,45 @@
+/* Copyright 2022 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+import {PluginType} from '../../metrics/types';
+import {TemporalScale} from '../histogram/histogram_types';
+import {Scale} from '../line_chart_v2/lib/public_types';
+
+export {Scale} from '../line_chart_v2/lib/public_types';
+
+export enum AxisDirection {
+  HORIZONTAL,
+  VERTICAL,
+}
+
+export enum Fob {
+  NONE,
+  START,
+  END,
+}
+
+export interface HistogramFobData {
+  type: PluginType.HISTOGRAMS;
+  scale: TemporalScale;
+  steps: number[];
+}
+
+export interface ScalarFobData {
+  type: PluginType.SCALARS;
+  scale: Scale;
+  minMax: [number, number];
```

Comment:
Can ScalarFobData also be modeled with `steps: number[]`? The Scalar charts are also a finite set of integer values. And in some environments the steps might be sampled so they aren't even a complete set of integers in the range [min,max].

Reply:
We thought about using the steps from each line. That seemed strange as each line can have different sets of steps. We could use a union of all the steps for each line. However, since there is no highlighting individual steps the continuous smooth dragging that we have implemented here makes the most sense. Lastly, the scale used in the Scalar cards requires this minMax(it is actually not the min and max steps that we have data for but, the min and max steps that are shown on the axis which is slightly different).

Reply:
> We could use a union of all the steps for each line.

I  mentioned offline that I think this particular idea is intriguing and possibly a nice compromise between behavior and complexity. I'd like us to try it out.